pkg/controller/job/job_controller_test.go

```diff
@@ -6594,19 +6594,21 @@ func TestWatchOrphanPods(t *testing.T) {
 					sharedInformers.Batch().V1().Jobs().Informer().GetIndexer().Delete(tc.job)
 				})
 			}
-
+			_, ctx := ktesting.NewTestContext(t)
```

Comment:
nit: Why is this test we call `NewTestContext` at both levels: the test and individual test cases? I looked at other tests and it seems we are generally creating the context either for the whole test or per test case, but we don't seem consistent: `TestSyncJobWithJobBackoffLimitPerIndex` is per test, while `TestSyncJobWithJobSuccessPolicy` is per test case.

Reply:
It looks like we need both, because there is code at both levels that need a context. And, in general, we want to associate any logs produced for a subtest to the subtest, not the parent test.

However, it looks odd that we are just overriding the context instead of creating a child of it.

Perhaps we need support from `ktesting` to do this better. Something like `log, ctx := ktesting.WithT(ctx, t)`.

@pohly do you have any other suggestion?

Reply:
I think it's okay to call `ktesting.NewTestContext` multiple times at different levels. There's nothing in the context of the parent context that the sub-test needs to inherit, is it? `ktesting.NewTestContext` itself puts a logger for `t` there which would just get overridden by a second call.

Reply:
> There's nothing in the context of the parent context that the sub-test needs to inherit, is it?

Not in this or other tests in this change, I think. But, it seems safer to inherit the context, specially if the parent context is canceled.

But we can leave it for a follow up.

Reply:
I see, thanks for explaining.